Feature/시간표 화면

BEEEAM-J opened the pull request:
## 📌 PR 요약

🌱 작업한 내용
- 시간표 화면 작업

🌱 PR 포인트
- "홈" 및 "시간표" 탭바 작업
- 시간표 상단 텍스트 바 수정
- 시간표 UI padding, radius 및 **시간 column width**,  **요일 Row Height** 수정

## 📸 스크린샷
|스크린샷|
|:--:|
|![Screenshot_20250730_234852](https://github.com/user-attachments/assets/0192b194-86ac-4266-9f0e-89c6a4e047a3)|

## 📮 관련 이슈
- Resolved: #13 

### RCA 룰을 사용하여 코드 리뷰를 해주세요
`R (Request Changes)` : 적극적으로 반영을 고려해주세요   
`C (Comment)` : 웬만하면 반영해주세요   
`A (Approve)` : 반영해도 좋고, 넘어가도 좋습니다. 사소한 의견입니다.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit


* **신규 기능**
  * 홈, 시간표, 마이페이지 탭이 포함된 새로운 탭 바가 상단에 추가되었습니다.
  * 마이페이지 아이콘 등 새로운 벡터 아이콘 리소스가 추가되었습니다.

* **기능 개선**
  * 시간표 화면 구조가 개선되어 탭 바와 앱바가 분리되고, 화면 상태에 따라 홈/시간표/빈 시간표 화면이 구분됩니다.
  * 시간표 셀 및 e-러닝 셀, 빈 셀의 모서리 라운딩 등 UI 스타일이 세분화되고 시각적으로 개선되었습니다.
  * 각 컴포넌트의 색상, 크기, 아이콘 스타일이 일관성 있게 변경되었습니다.

* **버그 수정**
  * 시간표의 금요일 및 e-러닝 셀 위치에 따른 모서리 스타일 적용 등 세부 UI 표시 오류가 수정되었습니다.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

BEEEAM-J committed:
Refactor: 시간표 화면 앱바 UI 수정

- 시간표 화면의 앱바 UI를 변경했습니다.
- 시간표 관련 아이콘의 크기, 색상, 내부 path를 수정했습니다.
- 년도와 학기 정보를 앱바에 표시하도록 추가했습니다.

BEEEAM-J committed:
feat: 시간표 화면 탭 바 기능 추가

- `CchTabBar` 컴포저블을 추가하여 홈, 시간표, 마이페이지 탭을 구현했습니다.
- 시간표 화면 상태(`TimetableScreenContent`)를 추가하여 현재 표시되는 내용을 관리합니다. (홈, 시간표, 빈 시간표)
- `TimetableScreen`에서 `CchTabBar`를 사용하여 화면 상단에 탭 바를 표시하고, 각 탭 클릭 시 해당 화면으로 전환되도록 로직을 수정했습니다.

BEEEAM-J committed:
feat: 시간표 UI 개선

- 시간표의 테두리 및 셀 스타일 변경
- 시간표 요일 헤더 높이 조정

lluke0 commented on composeApp/src/commonMain/kotlin/com/chukchukhaksa/mobile/common/designsystem/component/tabbar/CchTabBar.kt:
```diff
@@ -0,0 +1,90 @@
+    Icon(
+      modifier = Modifier
+        .padding(top = 4.dp, bottom = 4.dp, end = 10.dp)
+        .clickable { onClickMyPage() },
```
C: suwikiClickable을 사용해주시면 좋아요 (쓰로틀 효과가 적용되어있어요)

BEEEAM-J replied:
실수로 빼먹었네요...ㅎ 
감사합니다.

lluke0 commented on composeApp/src/commonMain/kotlin/com/chukchukhaksa/mobile/presentation/timetable/timetable/component/timetable/Timetable.kt:
```diff
@@ -40,44 +40,46 @@ fun Timetable(
             TimeColumn(
-                modifier = Modifier.weight(1f),
+                modifier = Modifier.weight(0.06f),
                 maxPeriod = maxPeriod,
+                isHasELearning = eLearningCellList.isNotEmpty(),
             )
 
             TimetableDay.entries
                 .sortedBy { it.idx }
                 .filter { it !in listOf(TimetableDay.SAT, TimetableDay.E_LEARNING) }
                 .forEach { day ->
                     ClassColumn(
-                        modifier = Modifier.weight(1f),
+                        modifier = Modifier.weight(0.188f),
```
C: 
val timeWeight = 0.06 
val classWeight = 1 - timeWeight / TimetableDay.entires.filterNot { it == E_LEARNING }.size

요런 느낌으로 하는건 어떠세요?

BEEEAM-J replied:
음.. 제 개인적인 취향은 코드 자체를 보는데 편해서 weight 값을 정적으로 정해놓는 것인데, 근데 수식으로 정리 하는게 확실히 깔끔할 거 같네요.

lluke0 commented on composeApp/src/commonMain/kotlin/com/chukchukhaksa/mobile/presentation/timetable/timetable/component/timetable/Timetable.kt:
```diff
@@ -40,44 +40,46 @@ fun Timetable(
                 ELearningCell(
                     onClickClassCell = onClickTimetableCell,
                     cell = cell,
+                    isLast = cell == eLearningCellList.flatMap { it.value }.last(),
```
A: eLearningCellList가 emptyList인 경우에 last()를 호출해도 괜찮던가요?

BEEEAM-J replied:
ELearningCell 컴포넌트는 eLearningCellList.forEach 함수 내에서 실행 되고 있습니다. 

eLearningCellList가 emptyList인 경우에는 forEach 가 호출 되지 않기 때문에 문제가 되지 않습니다.

lluke0 commented on composeApp/src/commonMain/kotlin/com/chukchukhaksa/mobile/common/designsystem/component/tabbar/CchTabBar.kt:
```diff
@@ -0,0 +1,90 @@
+  onClickHome: () -> Unit,
+  onClickTimetable: () -> Unit,
+  onClickMyPage: () -> Unit,
+) {
```
C: TimetableScreen이랑 CchTabBar랑 너무 강하게 엮여있어서 고민이네요. 근데 CchTabBar는 재활용될 것 같진 않아서 이대로 가도 될거같아요. 다만 CchTabBar대신 TimetableTabBar로 바꾸는건 어떠세요? CchTabBar는 공용으로 쓰일 느낌인데 지금 코드 상으로는 TimetableScreen 전용으로 보여서요!

BEEEAM-J replied:
저도 재사용은 안될 거 같아서 TimetableTabBar로 네이밍 변경 하는 것이 맞다고 생각이 드네요.

lluke0 commented on composeApp/src/commonMain/kotlin/com/chukchukhaksa/mobile/presentation/timetable/timetable/TimetableContract.kt:
```diff
@@ -23,3 +23,9 @@ sealed interface TimetableSideEffect {
   data object NavigateSemesterSelect : TimetableSideEffect
 }
+
+enum class TimetableScreenContent {
+  TIMETABLE,
+  EMPTY_TIMETABLE,
+  HOME
+}
```
A: 네이밍 변경 제안드립니다! TimetableTabType로 하는게 더 좋다고 생각해용 (큰 문제는 없어보이긴합니다.)

아래는 클로드 opus4 의견입니다.

일반적으로 탭 관련해서 `Content`라는 네이밍은 잘 사용하지 않습니다. 더 명확하고 관용적인 네이밍들을 사용하죠.

## 일반적으로 사용하는 네이밍

### 1. **Tab/TabType** (가장 일반적)
```kotlin
enum class Tab {
    HOME,
    TIMETABLE,
    MY_PAGE
}

// 또는
enum class TabType {
    HOME,
    TIMETABLE,
    MY_PAGE
}
```

### 2. **Screen/ScreenType**
```kotlin
enum class Screen {
    HOME,
    TIMETABLE,
    MY_PAGE
}
```

### 3. **NavigationItem** (Navigation Component 사용 시)
```kotlin
enum class NavigationItem {
    HOME,
    TIMETABLE,
    MY_PAGE
}
```

### 4. **Route** (Navigation 관련)
```kotlin
enum class Route {
    HOME,
    TIMETABLE,
    MY_PAGE
}
```

## Content를 사용하는 경우

`Content`는 보통 화면 내부의 콘텐츠 상태를 나타낼 때 사용합니다:

```kotlin
// 화면 내부 상태
sealed class ScreenContent {
    object Loading : ScreenContent()
    data class Success(val data: List<Item>) : ScreenContent()
    object Empty : ScreenContent()
    data class Error(val message: String) : ScreenContent()
}
```

현재 코드에서 `TimetableScreenContent`는 탭과 콘텐츠 상태가 섞여있어서 혼란스럽습니다. 탭을 나타내려면 `Tab`, `TabType`, `Screen` 등이 더 적절합니다.

BEEEAM-J replied:
"Tab", "TabType", "Screen" 이라는 네이밍은 너무 포괄적인 느낌이 강한 거 같아요.
시간표 화면과 관련이 있기 때문에 "TimetableScreen" 이라는 네이밍으로 수정하겠습니다.

BEEEAM-J committed:
Refactor: Timetable 레이아웃 가중치 계산 방식 변경

- 요일별 수업 열의 가중치를 동적으로 계산하도록 변경하여, 토요일과 E-러닝을 제외한 요일 수에 따라 균등하게 분배되도록 수정했습니다.

BEEEAM-J committed:
refactor: CchTabBar -> TimetableTabBar

- `CchTabBar`와 `CchTabBarItem` 컴포저블을 각각 `TimetableTabBar`와 `TimetableTabBarItem`으로 이름을 변경했습니다.
- `TimetableScreenContent` enum을 `TimetableScreen`으로 이름을 변경하고 관련 로직을 수정했습니다.

BEEEAM-J committed:
refactor: 주석 처리된 코드 제거

- timetableBorderWidth 변수의 주석 처리된 값을 삭제합니다.

BEEEAM-J committed:
Merge branch 'develop' into feature/#13-timetable-ui

# Conflicts:
#	composeApp/src/commonMain/kotlin/com/chukchukhaksa/mobile/presentation/timetable/timetable/TimetableScreen.kt
#	composeApp/src/commonMain/kotlin/com/chukchukhaksa/mobile/presentation/timetable/timetable/component/TimetableAppbar.kt
#	composeApp/src/commonMain/kotlin/com/chukchukhaksa/mobile/presentation/timetable/timetable/component/timetable/cell/ELearningCell.kt

BEEEAM-J committed:
refactor: 컴포넌트 이름 변경 및 테마 적용

- `ChukChuk` 접두사를 `Cch`로 변경
  - `ChukChukRegularTextField` -> `CchRegularTextField`
  - `ChukChukAppBarWithTitle` -> `CchAppBarWithTitle`
  - `ChukChukSelectionContainer` -> `CchSelectionContainer`
- `CCHaksaTheme`를 `CchTheme`로 변경하여 테마 적용
- `suwikiClickable`을 `cchClickable`로 변경
- 불필요한 import 제거
